Fixes on several reps YAML files

antonvino opened the pull request:
Have done some testing with dogooderapp and have fixed representative files that failed to get through.
Fixes included:
- State options (some were empty)
- ABBREV => FULL for state in some files
- Removal of state/city/zip5/zip4 on the second form step when it's not already required (usually after the 1st step)
- Some outdated forms i.e. when enews popup is not applicable anymore

j-ro commented:
Thanks for posting a PR! Unfortunately, I'm not sure we can merge this in its current state. 

For example, the changes that have been made to H001052 would take this YAML from successful to failing. Address is indeed required, but it has been made non-required. We generally indicate we'd like a response when asked, but that has been removed. The success condition has been turned into an empty string, which means it is not actually checking for a real success message, and thus anything would pass even if the form did not submit.

In S001156, the replacements of empty options with states, the form actually does not have these state options. David Scott just has an option menu that's simply Georgia. So I suppose we could remove the entire select, but adding in options that don't exist in the DOM on the form tends to cause errors.

I haven't looked through the rest.

Can you look through your changes and make updates to ensure they're successfully filling out forms? Since folks pull directly from this repo for their YAMLs we have fairly stringent standards for making changes.

antonvino commented:
Sure, will re-check them again keeping your notes in mind.

antonvino commented:
On another note:
U000031 - gives a mobile version when a robot visits, so there is no way to retract the DOM elements (there are none for mobile version). 
See screenshot by the robot: 
![contactrobot-1270-62085-1431317453 46](https://cloud.githubusercontent.com/assets/11321765/7558464/42a80846-f7e8-11e4-8561-37745228fc30.png)

Is there any way to prevent that?


Also in some forms I think the script automatically does a find on selector which does not do much. For instance: 
H001063
- find:
-        - selector: "h1"
-          value: "Email Me- Thank You"

It does so before the success check so does not really check therefore is not needed or am I wrong?

j-ro commented:
Re: mobile, I don't have issues with upton with our implementation, so it may be your robot.

Re: find, yes, that's necessary. Sometimes forms submit via javascript or other means, so sometimes they take a while with lots of content, so find is necessary to wait for the success page to show up before we check the success condition.

antonvino commented:
Hi. I have reviewed the changes and it should be all good now.

Just FYI:
members/B001275.yaml
The original form on Bucshon's website has empty " " for values in the TOPIC select, yet the previous YAML file had values in the " ". I'm not sure what to do there.

For most of them removing state/city/zip5&4 when they were not required on the second step took them to success for me.
Let me know if there are any more issues with this PR.

j-ro commented:
Still some issues I'm afraid. Some examples:

B001275, the change in state abbreviations to full isn't correct -- abbreviations are the values that are used in fact, though the labels are full.

In B001275, generally we don't add in non required fields, or at least we're slowly trying to stop that. So adding in address2 probably shouldn't happen. But what you did with the select seems right. Weird markup!

H001052, zip4 is probably not always required, but probably required sometimes to put the user in the right district (and thus allow access to the form). So this should be marked as true.

Haven't gone all the way through yet, but just some examples.

antonvino commented:
Ok. 
True on State abbrev, the full state in the select threw me off. Fixed.
Have put ZIP4 back to required:true 
I have removed the non-required fields where possible in my fixes.
Left the phone fields in for M000702 even if not required, since there is a required one and they are all copied in (I presume).

j-ro commented:
Great, getting there!

In M000702, probably don't need to select twice the topic -- you do it at the beginning, but then don't need to do it again.

S001156, watch your indents

I think that might be it!

antonvino commented:
I've fixed those on Friday, just to let you know.

j-ro commented:
great, merging these over -- would be good to confirm these are all tested and working in production when you can!

DrMeers commented on members/B001275.yaml:
```diff
@@ -71,31 +55,31 @@ contact_form:
-            Trade: "Trade"
-            Veterans: "Veterans"
-            Other: "Other"
+            Abortion: " "
```
Why were these blanked out @antonvino ?

j-ro replied:
Take a look at the form, that's how they are on the page.

DrMeers replied:
Oh wow, I see. Apparently Larry really doesn't care what topic you pick?!

j-ro replied:
This isn't even the craziest/stupidest markup I've seen on these things. It's a jungle out there!
